src/test/java/uk/gov/homeoffice/digital/sas/balancecalculator/testutils/CommonUtils.java

```diff
@@ -129,4 +133,18 @@ public static  <T> T loadObjectFromFile(String filePath, Class<T> type) throws I
 
     return mapper.readValue(file, type);
   }
+
+  public static void assertTypeAndDateAndTotals(Accrual accrual, AccrualType expectedAccrualType,
+                                                LocalDate expectedAccrualDate,
```

Comment:
To simplify a little bit further, the type could be changed to String, meaning that the code using this method will only have to pass on the expected accrual date like this for example: `2023-04-01`.
Line 142 would have to be adjusted: `assertThat(accrual.getAccrualDate().toString()).isEqualTo(expectedAccrualDate)`

Reply:
Update the signature to use a String. 
SonarLint advised using the hasToString() instead of using .toString().